Tools/Scripts/webkitpy/port/base.py

```diff
@@ -134,7 +143,10 @@ def __init__(self, host, port_name, options=None, **kwargs):
         self._options = options or optparse.Values()
 
         if self._name and '-wk2' in self._name:
-            self._options.webkit_test_runner = True
+            self._options.driver_names = [self.DRIVER_NAMES[0]]
+
+        if not self.driver_name().startswith(self.DRIVER_NAMES):
+            raise ValueError(f'{self.driver_name()} is not supported on this platform.')
```

Comment:
Nit: I'd probably be tempted to raise a custom exception type here, something like `UnsupportedDriverError`.